base/inference.jl

```diff
@@ -2223,16 +2161,17 @@ function effect_free(e::ANY, sv, allow_volatile::Bool)
         allow_volatile && return true
         return isconst(e.mod, e.name)
     end
-    if isconstantref(e, sv) !== false
-        return true
-    end
+    # TODO restore this
+    #if isconstantref(e, sv) !== false
+    #    return true
+    #end
     if isa(e,Expr)
         e = e::Expr
         if e.head === :static_typeof
             return true
         end
         ea = e.args
-        if e.head === :call
+        if e.head === :call && !isa(e.args[1], GenSym) && !isa(e.args[1], Slot)
```

Comment:
This is related to a bug I discovered in `inlineable`. We call `effect_free` on expressions from the inlined function, but passing the `InferenceState` of the enclosing function. Then it tries to look up out-of-range slots, etc. Excluding slots and gensyms here is a temporary workaround.